core/src/main/scala/org/apache/spark/h2o/converters/H2ORDD.scala

```diff
@@ -54,22 +55,30 @@ class H2ORDD[A <: Product: TypeTag: ClassTag, T <: Frame] private(@(transient @p
 
   def mkString(seq: Seq[_], sep: Any) = if (seq == null) "(null)" else seq.mkString(sep.toString)
 
-  val colNamesInFrame = frame.names()
+  val colNames = frame.names()
+
+  override val selectedColumnIndices: Array[Int] = colNames.indices.toArray
+
+  override val expectedTypes: Option[Array[Byte]] = {
+    // there is no need to prepare expected types in internal backend
+    if (isExternalBackend) {
+      // prepare expected types for selected columns
+      Option(ExternalFrameUtils.prepareExpectedTypes(productType.types))
+    } else {
+      None
+    }
+  }
 
   // Check that H2OFrame & given Scala type are compatible
-  if (!productType.isSingleton) {
+  if (!productType.isSingleton) { // Unnecessary check
```

Comment:
@vpatryshev It's not related to this PR so I will remove this comment here. But do we need this check here in the future ? The reason I could see why it's needed is that in case when `Product` `RDD` has only one field, it doesn't matter on it's and corresponding Frame name and we don't compare them as we do in cases where there is 2 and more fields in the RDD. Was that the intention ? If yes, I guess it make sense to have it like this, but I would put a comment in the code explaining this behaviour.